tests/compile-fail/bool_comparison.rs

```diff
@@ -0,0 +1,23 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#[deny(bool_comparison)]
+fn main() {
+    let x = true;
+    if x == true { "yes" } else { "no" };
+    //~^ ERROR equality checks against booleans are unnecesary
+    //~| HELP try simplifying it:
+    //~| SUGGESTION x
+    if x == false { "yes" } else { "no" };
+    //~^ ERROR equality checks against booleans are unnecesary
+    //~| HELP try simplifying it:
+    //~| SUGGESTION !x
+    if true == x { "yes" } else { "no" };
+    //~^ ERROR equality checks against booleans are unnecesary
+    //~| HELP try simplifying it:
+    //~| SUGGESTION x
+    if false == x { "yes" } else { "no" };
+    //~^ ERROR equality checks against booleans are unnecesary
+    //~| HELP try simplifying it:
+    //~| SUGGESTION !x
```

Comment:
Let's keep the whole suggestion output here (just run with `rustc tests/compile-fail/bool_comparison.rs
 -L target/debug` to see the suggestion). 

In this case, `SUGGESTION if !x { "yes" } else { "no" };`. Helps protect against bugs in span_suggestion.